core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineStrategy.java

```diff
@@ -0,0 +1,48 @@
+package com.scalar.db.storage.jdbc;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.io.DataType;
+import com.scalar.db.storage.jdbc.query.QueryUtils;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+abstract class RdbEngineStrategy {
+
+  static RdbEngineStrategy create(JdbcConfig config) {
+    switch (config.getRdbEngine()) {
+      case MYSQL:
```

Comment:
[trivial] A future potential improvement might be like the following so that we don't need to update this file

```java
public enum RdbEngine {
  MYSQL(RdbEngineMysql::new),
  POSTGRESQL(RdbEngineMysql::new),
  ORACLE(RdbEngineMysql::new),
  SQL_SERVER(RdbEngineMysql::new);

  private final Supplier<RdbEngineStrategy> factory;

  RdbEngine(Supplier<RdbEngineStrategy> factory) {
    this.factory = factory;
  }
  
  RdbEngineStrategy createStrategy() {
    return factory.get();
  }
}
```

Reply:
Similar to the above comment and it's a minor thing, but I feel it might be slightly better to separate the creation of `RdbEngineStrategy` instance and its interface from the "separation of concerns" perspective.

It depends on how `RdbEngineStrategy` is going to be extended, but we can make it just an interface instead of the abstract class by extracting the `create` method and making `enclose` default method.

Reply:
@feeblefakie I fully agree with the idea that we can (and should) make `RdbEngineStrategy` interface rather than `abstract class`.

I did it in another pull-request (will ask your review after the base branch is merged to master).
https://github.com/scalar-labs/scalardb/pull/791/files#diff-f2858564c4aa48bdf1bc09aa29ebb2f9def9f18245b75de17415d4f3ac36a261

Reply:
Sounds good to me! Thanks!